### PR TITLE
Memleak fixes - Part II.

### PR DESCRIPTION
**Pull request by UltraNix**
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Battlefield/Battlefield.cpp [hunk 1]
```diff
@@ -814,6 +812,7 @@ Creature* Battlefield::SpawnCreature(uint32 entry, float x, float y, float z, fl
 
     // Set creature in world
     map->AddToMap(creature);
+    creature->setActive(true);
```
Comment (UltraNix) [inline src/server/game/Battlefield/Battlefield.cpp:815]: Not sure about it.
[valgrind (2).log](https://github.com/azerothcore/azerothcore-wotlk/files/6450460/valgrind.2.log)
There is a lot of memleaks related to that function, but not sure what is the cause. This is only my suspicion.

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -361,14 +361,14 @@ ObjectMgr* ObjectMgr::instance()
     return &instance;
 }
 
-void ObjectMgr::AddLocaleString(std::string const& s, LocaleConstant locale, StringVector& data)
+void ObjectMgr::AddLocaleString(std::string&& s, LocaleConstant locale, StringVector& data)
```
[the remark this answers is not in the document]
Comment (UltraNix) [reply src/server/game/Globals/ObjectMgr.cpp:364]: https://stackoverflow.com/questions/5481539/what-does-t-double-ampersand-mean-in-c11

[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/server/game/Globals/ObjectMgr.cpp:364]: interesting, thanks

**Comment by FrancescoBorzi** [comment]
@UltraNix the build is failing :(

**Comment by Yehonal** [comment]
This could help to spot memory leak: https://github.com/azerothcore/azerothcore-wotlk/pull/5769

**Comment by Branel** [comment]
To test this I just need to have a server running and make sure it doesnt crash after a certain amount of time?

**Comment by UltraNix** [comment]
Tested on @mpfans server and it got better.

**Comment by mpfans** [comment]
> To test this I just need to have a server running and make sure it doesnt crash after a certain amount of time?

Sorry, I forgot to give feedback here
I tested one day and didn't crash
This PR further improves the memory leak

**Pull request merged**